main.go

```diff
@@ -313,6 +308,12 @@ func Main(configPath string, configTest bool, buildVersion string) {
 	attachCommands(ssh, hostMap, handshakeManager.pendingHostMap, lightHouse, ifce)
 	ifce.Run(config.GetInt("tun.routines", 1), udpQueues, buildVersion)
 
+	// Start DNS server last to allow using the nebula IP as lighthouse.dns.host
+	if amLighthouse && serveDns {
```

Comment:
It would be great if the reload support also worked for the `lighthouse.serve_dns` and `lighthouse.dns.enabled` config options to turn on/off the dns server with HUP.

Reply:
Good point. I'll try to get that in today.

Reply:
Hmm, this is actually tricky because dns.Server.isStarted is not exported in miekg/dns, so we cannot easily figure out whether a DNS server is running without keeping state about that within nebula as well.

Without this we would have to restart the DNS server on every HUP, even when no config option has changed, simply because there's no way to figure out whether it's already running or needs to be started.

Reply:
hmm, not the end of the world. I think the next move on the dns server is to get it out of leaking itself onto the global scope like it does. Would then lend itself to keeping that state.